ALUM/ALUM/Views/SignUpPage/SignUpPageViewModel.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  SignUpPageViewModel.swift
+//  ALUM
+//
+//  Created by Neelam Gurnani on 2/17/23.
+//
+
+import Foundation
+import FirebaseCore
+import FirebaseAuth
+
+final class SignUpPageViewModel: ObservableObject {
+    @Published var passwordAgain: String = ""
+    @Published var disabled: Bool = true
+    @Published var emailFunc: [(String) -> (Bool, String)] = []
+    @Published var passFunc: [(String) -> (Bool, String)] = []
+    @Published var passAgainFunc: [(String) -> (Bool, String)] = []
+    @Published var isMentee = false
+    @Published var isMentor = false
+    @Published var setUpIsInvalid = false
+    
+    @Published var account = Account(name: "", email: "", password: "")
+    @Published var mentee = Mentee(name: "", email: "", grade: "", mentorshipGoal: "", password: "")
+    @Published var mentor = Mentor(name: "", email: "", yearOfGrad: "", university: "", major: "", minor: "", intendedCareer: "", password: "")
+    
+    func setUpMentee() {
+        mentee.name = account.name
+        mentee.email = account.email
+        mentee.password = account.password
+    }
+    
+    func setUpMentor() {
+        mentor.name = account.name
+        mentor.email = account.email
+        mentor.password = account.password
+    }
+    
+    
+    func checkPasswordSame() {
+        if !(self.account.password == self.passwordAgain) {
+            self.passAgainFunc = [SignUpPageViewModel.Functions.passNotSame]
+        } else {
+            self.passAgainFunc = []
+        }
+    }
+    
+    class Functions {
```

Comment:
action - Let's move this class out of ViewModel. Instead make a class called  `SignUpFlowErrorFunctions` and store that in path - 
`ALUM/Constants/ErrorFunctions/SignUpFlowErrorFunctions.swift`

Since this functions don't depend on any values from the `SignUpModel` it doesn't need access to the ViewModel object so by having it inside `ViewModel` we are only cluttering it. Moving it to a ErrorFunctions directory will help us keep all these functions at one place.